### PR TITLE
Bump go, controller-runtime and flux components versions

### PR DESCRIPTION
**Pull request by aryan9600**
Signed-off-by: Sanskar Jaiswal <sanskar.jaiswal@weave.works>

### CODE REVIEW

File: controllers/gitrepository_watcher.go [hunk 1]
```diff
@@ -43,7 +43,10 @@ type GitRepositoryWatcher struct {
 // +kubebuilder:rbac:groups=source.toolkit.fluxcd.io,resources=gitrepositories/status,verbs=get
 
 func (r *GitRepositoryWatcher) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	log := logr.FromContext(ctx)
+	log, err := logr.FromContext(ctx)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to initialize logger, error: %w", err)
+	}
```
Comment (relu) [inline controllers/gitrepository_watcher.go:49]: Can you please change this to `log := ctrl.LoggerFrom(ctx)`?

**Review by relu — CHANGES_REQUESTED** [review]

**Review by relu — CHANGES_REQUESTED** [review]
Please also:
- [ ] Update the `controller-gen` tool to `0.7.0` in the [Makefile](https://github.com/fluxcd/source-watcher/blob/main/Makefile#L74)
- [ ] Update the [Dockerfile](https://github.com/fluxcd/source-watcher/blob/main/Dockerfile#L1) base image go version
- [ ] Update the [Dockerfile](https://github.com/fluxcd/source-watcher/blob/main/Dockerfile#L19) alpine version to 3.15
- [ ] Update the GitHub Actions [test workflow file](https://github.com/fluxcd/source-watcher/blob/main/.github/workflows/test.yaml#L25) go-version

Thank you for your contribution!

**Comment by stefanprodan** [comment]
Please rebase with main, I've merged your Makefile changes.

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @aryan9600 for the updates and thanks @relu for guidance

**Pull request merged**